deepchem/feat/base_classes.py

```diff
@@ -207,6 +208,198 @@ def __call__(self, molecules):
     return self.featurize(molecules)
 
 
+class StructureFeaturizer(Featurizer):
+  """
+  Abstract class for calculating a set of features for an
+  inorganic crystal structure.
+
+  The defining feature of a `StructureFeaturizer` is that it
+  operates on 3D crystal structures with periodic boundary conditions. 
+  Inorganic crystal structures are represented by Pymatgen structure
+  objects. Featurizers for inorganic crystal structures that are subclasses of
+  this class should plan to process input which comes as pymatgen
+  structure objects. 
+
+  This class is abstract and cannot be invoked directly. You'll
+  likely only interact with this class if you're a developer. Child 
+  classes need to implement the _featurize method for calculating 
+  features for a single crystal structure.
+
+  Notes
+  -----
+  Some subclasses of this class will require pymatgen and matminer to be
+  installed.
+
+  """
+
+  def featurize(self, structures: Iterable[dict],
```

Comment:
I think it is better to clarify JSON type like this.


```
JSON = Dict[str, Any]

def featurize(self, structures: Iterable[JSON],
```

Reply:
Thanks @nd-02110114 ! I updated the type.